### PR TITLE
Users API

### PR DESCRIPTION
**Pull request by dickymr**
Changes:
1. Renamed 'auth' to 'user' in preparation for adding an admin schema later.
2. Updated the validate middleware to support validation for body, params, and query.
3. Created the isAuthenticated middleware.
4. Added /users/profile endpoint to retrieve the profile of the authenticated user.

Please review. Thank you!

### CODE REVIEW

File: server/helpers/index.ts [hunk 1]
```diff
@@ -19,4 +19,9 @@ export const createToken = (userId: string, email: string): string => {
     const token = jwt.sign(payload, 'supersecretkey', { expiresIn: '1h' });
```
Comment (jeisaRaja) [inline server/helpers/index.ts]: I think secret key better be stored inside .env

File: server/helpers/index.ts [hunk 2]
```diff
@@ -19,4 +19,9 @@ export const createToken = (userId: string, email: string): string => {
...
 
     return token;
-};
\ No newline at end of file
+};
+
+export const verifyToken = (token: string): string | JwtPayload => {
+    const decodedToken =  jwt.verify(token, 'supersecretkey');
```
Comment (jeisaRaja) [inline server/helpers/index.ts]: this one also

**Review by jeisaRaja — COMMENTED** [review]
Looks good to me, only suggestion is to move secret key to .env

Also, I think we need to add cors, because I got cors error when trying to send request.
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8080/signup. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.`

**Comment by dickymr** [comment]
> Looks good to me, only suggestion is to move secret key to .env
> 
> Also, I think we need to add cors, because I got cors error when trying to send request.
> `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8080/signup. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.`

Got it, I'll do it by tonight

**Comment by dickymr** [comment]
> Looks good to me, only suggestion is to move secret key to .env
> 
> Also, I think we need to add cors, because I got cors error when trying to send request.
> `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8080/signup. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 200.`

Updated in the last two commits. Please check

**Review by jeisaRaja — APPROVED** [review]
Nice, I tested it and everything looks good, thank you.

**Pull request merged**